File: easy/2078-two-furthest-houses-with-different-colors/solution.java
/**
 * 2078. Two Furthest Houses With Different Colors
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public int maxDistance(int[] colors) {
        int n = colors.length;
        int dist = 0;

        // Check from first house - find furthest different color
        for (int j = n - 1; j >= 0; j--) {
            if (colors[j] != colors[0]) {
                dist = Math.max(dist, j);
                break;
            }
        }

        // Check from last house - find furthest different color
        for (int i = 0; i < n; i++) {
            if (colors[i] != colors[n - 1]) {
                dist = Math.max(dist, n - 1 - i);
                break;
            }
        }

        return dist;
    }
}

File: easy/2078-two-furthest-houses-with-different-colors/solution.py
from typing import List


class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        """
        2078. Two Furthest Houses With Different Colors
        Time: O(n)
        Space: O(1)
        """
        n = len(colors)
        dist = 0

        # Check from first house - find furthest different color
        for j in range(n - 1, -1, -1):
            if colors[j] != colors[0]:
                dist = max(dist, j)
                break

        # Check from last house - find furthest different color
        for i in range(n):
            if colors[i] != colors[n - 1]:
                dist = max(dist, n - 1 - i)
                break

        return dist
